cluster/network.go

```diff
@@ -92,9 +92,19 @@ func (networks Networks) Filter(filter filters.Args) Networks {
 				}
 			}
 		}
+		// substring match based on name filter
+		for _, idOrName := range append(names, ids...) {
+			if networkList := networks.substringMatch(idOrName); len(networkList) != 0 {
```

Comment:
I'm not positive, but I believe you can just do `filter.Match("name", network.Name)` here, which is what the engine does:

https://github.com/moby/moby/blob/master/api/server/router/network/filter.go#L48